tests/browser/index.test.js

```diff
@@ -310,6 +310,35 @@ suite('<Entity events/>', () => {
     });
   });
 
+  test('can remove event listener when component is detached', function (done) {
+    let bazHandlerCalled = false;
+    function bazHandler () { bazHandlerCalled = true }
+
+    ReactDOM.render(
+      <Scene>
+        <Entity id='foo'>
+          <Entity id='bar' events={{baz: bazHandler}}/>
+        </Entity>
+      </Scene>,
+      div
+    );
+    div.querySelector('a-scene').addEventListener('loaded', () => {
+      ReactDOM.render(
+        <Scene>
+          <Entity id='foo'/>
+        </Scene>,
+        div
+      );
+      div.querySelector('a-entity').emit('baz');
```

Comment:
I have filed an issue to make what I experienced more explicit:
https://github.com/aframevr/aframe-react/issues/92

The call to remove events added by this PR does fix my issue. But maybe there is something else at play I have not understood.